### PR TITLE
Kimberley Mackenzie & Kayla Ecker - Carets

### PR DESCRIPTION
**Pull request by kimberleyamackenzie**
# Scrabble
Congratulations! You're submitting your assignment.

## Comprehension Questions

|  Feature 	|   Feedback	|
|---	|---	|
|   Why do you think we chose to make the `score` method in the `Scoring` class a **class method** instead of an **instance method**? |  If the score method was an instance method, a new instance of score would have to be created or changed each time a word needed to be scored, which would be unnecessary & repetitive.    |
|   Describe an example of a test you wrote for a _nominal case_. |   A nominal case would be our test for highest scoring word in the Scoring class.  We used the following array ["cat", "dog", "cats"] - in this case, one of the words was the clear winner ("cats"), and there were no ties, which is the sort of situation you would expect most of the time.
	|
|   Describe an example of a test you wrote for an _edge case_. |   For an edge case of the same method, highest scoring word in Scoring class, we input an array of the following words  ["bob", "zzzz", "zzkkkk","zzzdddf","billy"].  In this case, there was a tie, but one of the words was 7 letters, which meant it should automatically win- a situation that would probably occur rarely, but is possible.
	|
|   What was your final test coverage percentage?	| 100%   	|
|   Is there a specific piece of code you'd like feedback on?	|  We'd love to hear feedback on our tests, as well as on the optional requirement we didn't implement (but left hashed out) because we had trouble implementing the rest of our tests to work with it - if we had figured out the testing side, would it have worked?    	|

### CODE REVIEW

**Comment by CheezItMan** [comment]
What optional enhancement that you worked on do you want to get feedback on?

File: lib/scoring.rb [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        ['A', 'E', 'I', 'O', 'U', 'L', 'N', 'R', 'S', 'T']=> 1,
+        ['D', 'G'] =>	2,
+        ['B', 'C', 'M', 'P'] =>	3,
+        ['F', 'H', 'V', 'W', 'Y'] =>	4,
+        ['K'] =>	5,
+        ['J', 'X'] =>	8,
+        ['Q', 'Z'] => 10
+      }
+
+      letters_only = /^[a-zA-Z]{1,7}$/
```
Comment (CheezItMan) [inline lib/scoring.rb:15]: Good use of regex.

File: lib/player.rb [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      drawn_tiles = tile_bag.draw_tiles(num)
+      num.times do
+        @tiles << drawn_tiles.pop
+      end
+    end
+
+    private
+
+    def won?
+      return total_score > 100 ? true : false
```
Comment (CheezItMan) [inline lib/player.rb:61]: could also be

`return total_score > 100`

File: lib/tilebag.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    end
+
+    def draw_tiles(num)
+      if num == 0 || num > 7 || num.class != Integer
+        raise ArgumentError.new("This is incorrect. Enter a number between 1 and 7.")
+      end
+
+      tiles_array = []
+      num.times do
+        @tiles.shuffle!
```
Comment (CheezItMan) [inline lib/tilebag.rb:30]: move `@tiles.shuffle!` outside the loop.  Otherwise you're shuffling over and over and over again.

**Comment by CheezItMan** [comment]
# Scrabble
## What We're Looking For

|  Feature  |   Feedback |
|---	|---	|
|  **General**	|   	|
|   Answered comprehension questions  | Check, I'll look at your tests.   |
|   Both Teammates contributed to the codebase	|  Chcek  |
|   Regular Commits with meaningful commit messages  | Good # of commits and good commit messages   |
|  **Scoring class**	|   	|
|  `score` method	|  Check  |
|   Uses appropriate data structure to store the letter score	| Check   |
|   Appropriately handles edge cases	|  Good use of Regex.    |
|   Tests `score` edge cases (one-letter word, 7 letter word)	| No test of an empty string for a word.     |
|  `highest_scoring_word` method	|  Check |
|   Appropriately handles edge cases	|  Well done  |
|   Tests `highest_scoring_word` edge cases (ties, no words, no array)	|  Nice work, but I would also vary the order of your word array and verify that it gives the correct answer.     |
|  **Player class**	|   	|
|  Uses an array to keep track of words played	|  Check  |
|  Uses existing code to create `highest_scoring_word` and `highest_word_score`	|  Check  |
|  Returns `true` or `false` from `won?` method | Check  |
|  Tests edge cases on the `play(word)` method for words that cannot be scored |   Check, good tests of empty plays, 1 word plays and plays with 7-letter words, and testing for the highest scoring word on a tie being the shorter word.  |
|  **TileBag class**	|   	|
|  Uses a data structure to keep the set of default tiles in the bag	|  Nice use of a CSV!    |
|  `draw_tiles` method uses the parameter to draw a set of random tiles |  Check, good checks on the type of the parameter as well.  |
| `tiles_remaining` method returns a count	| Check   |
|  Tests `draw_tiles` edge cases for different parameter inputs	|  Check, but you're not testing to see if they draw more tiles than remain!  |
|  Summary  |  Nice work.  You've got some edge cases that you need to cover, but you did a good job overall.  I left a few notes in your code as well.  |

**Review by CheezItMan — COMMENTED** [review]
Just a few comments
